velox/exec/tests/RowContainerTest.cpp

```diff
@@ -284,121 +285,304 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
       assertEqualVectors(expected, result);
     }
-    // Verify compareRows method with default compare flags.
-    if (ascending && nullsFirst) {
-      std::sort(
-          indexedRows.begin(),
-          indexedRows.end(),
-          [&](const std::pair<int, char*>& l, const std::pair<int, char*>& r) {
-            return rowContainer->compareRows(l.second, r.second) < 0;
-          });
-      std::vector<std::optional<T>> sorted;
-      for (const auto& irow : indexedRows) {
-        if (decoded.isNullAt(irow.first)) {
-          sorted.push_back(std::nullopt);
-        } else {
-          sorted.push_back(decoded.valueAt<T>(irow.first));
-        }
+  }
+
+  template <typename T, typename V>
+  void testOrderAndNullsFirstVariations(
+      const TypePtr& type,
+      const VectorPtr& values,
+      const std::vector<std::optional<V>>& ascNullsFirstOrder,
+      std::function<VectorPtr(const std::vector<std::optional<V>>&)>
+          generateExpectedVector) {
+    // The default flags are ascending == true, nullsFirst == true.
+    // std::nullopt means use the default for the compare function.
+    const std::vector<std::optional<CompareFlags>> compareFlags{
+        {{std::nullopt}}, {{true, false}}, {{false, true}}, {{false, false}}};
+
+    for (auto flags : compareFlags) {
+      std::vector<std::optional<V>> expectedOrder{ascNullsFirstOrder};
+      // The expectedOrder is ascNullsFirstOrder for the case
+      // where no compare flags are provided.
+      if (flags.has_value()) {
+        changeSortingOrder<V>(expectedOrder, flags.value());
       }
-      VectorPtr result = vectorMaker.template flatVectorNullable<T>(sorted);
-      assertEqualVectors(expected, result);
+      auto expected = generateExpectedVector(expectedOrder);
+
+      testRowContainerCompareAPIs<T>(type, values, expected, flags);
+    }
+  }
+
+#define TEST_FLOATING_TYPE_LIMIT_VARIABLES        \
+  auto lowest = std::numeric_limits<T>::lowest(); \
+  auto min = std::numeric_limits<T>::min();       \
+  auto max = std::numeric_limits<T>::max();       \
+  auto nan = std::numeric_limits<T>::quiet_NaN(); \
+  auto inf = std::numeric_limits<T>::infinity()
+
+  template <typename T>
+  void testComparePrimitiveFloats(const TypePtr& type) {
+    // NOTE: set 'isJoinBuild' to true to enable nullable sort key in test.
+    TEST_FLOATING_TYPE_LIMIT_VARIABLES;
+
+    auto values = makeNullableFlatVector<T>(
+        {std::nullopt, max, inf, 0.0, nan, lowest, min});
+
+    std::vector<std::optional<T>> ascNullsFirstOrder = {
+        std::nullopt, lowest, 0.0, min, max, inf, nan};
+
+    testOrderAndNullsFirstVariations<T, T>(
+        type, values, ascNullsFirstOrder, [&](const auto& expectedOrder) {
+          return makeNullableFlatVector<T>(expectedOrder);
+        });
+  }
+
+  template <typename T>
+  void testCompareArrayFloats(const TypePtr& type) {
+    // NOTE: set 'isJoinBuild' to true to enable nullable sort key in test.
```

Comment:
What does this note applies to? It is not clear.

Reply:
This comment was present in the previous code that was refactored.

Reply:
Upon closer, look, the `RowContainer` has an `isNullableKeys` option and the tests turns that off (using a variable called `isJoinBuild` in the wrapper function). Which is interesting because the did have null values in the input.